CMake/external_libcurl.cmake

```diff
@@ -0,0 +1,33 @@
+if(ENABLE_RS_AUTO_UPDATER)
```

Comment:
"RS" really needed?
Is this the official name for the feature? Auto updater?

Reply:
We can drop the RS.
about the name that's the name I know for this feature.
Now that I am thinking about it maybe we should separate it into 2 flags.
1 for File downloading (libcurl + http-downloader)
1 for auto updates
It is reasonable that we can use a file downloader capability without the binding to auto updates feature.
What do you think?

Reply:
I think that's probably wise :)
The thing about the name: what exactly is "auto" about it? It requires user input, so I think a name like "software update" makes more sense (SW_UPDATE?)..